spec/index.bs

```diff
@@ -6493,29 +6476,38 @@ dictionary GPURenderPipelineDescriptor : GPUPipelineDescriptorBase {
+        1. If the `front_facing` [=builtin=] is an input of |fragment|:
             1. Decrement |maxFragmentShaderInputComponents| by 1.
-        1. If the `sample_index` [=builtin=] is an input of |descriptor|.{{GPURenderPipelineDescriptor/fragment}}:
+        1. If the `sample_index` [=builtin=] is an input of |fragment|:
             1. Decrement |maxFragmentShaderInputComponents| by 1.
-        1. If the `sample_mask` [=builtin=] is an input of |descriptor|.{{GPURenderPipelineDescriptor/fragment}}:
+        1. If the `sample_mask` [=builtin=] is an input of |fragment|:
             1. Decrement |maxFragmentShaderInputComponents| by 1.
-    1. Return `true` if and only if all of the following conditions are satisfied:
-        - The [=location=] of each user-defined output of |descriptor|.{{GPURenderPipelineDescriptor/vertex}} is less
+    1. Return `true` if and only if all of the following requirements are met:
+        - The [=location=] of each user-defined output of |vertex| must be less
             than (|device|.limits.{{supported limits/maxInterStageShaderVariables}}).
-        - The [=location=] of each user-defined input of |descriptor|.{{GPURenderPipelineDescriptor/fragment}} is less
+        - The [=location=] of each user-defined input of |fragment| must be less
             than (|device|.limits.{{supported limits/maxInterStageShaderVariables}}).
-        - There are no more than |maxVertexShaderOutputComponents| scalar
-            components across all user-defined outputs for
-            |descriptor|.{{GPURenderPipelineDescriptor/vertex}}.
+        - There must be no more than |maxVertexShaderOutputComponents| scalar
+            components across all user-defined outputs for |vertex|.
             (For example, a `f32` output uses 1 component, and a `vec3<f32>` output uses 3 components.)
-        - There are no more than |maxFragmentShaderInputComponents| scalar
-            components across all user-defined inputs for
-            |descriptor|.{{GPURenderPipelineDescriptor/fragment}}.
+        - There must be no more than |maxFragmentShaderInputComponents| scalar
+            components across all user-defined inputs for |fragment|.
+        - In order by [=location=], the user-defined inputs of |fragment| must be an exact prefix of
+            the user-defined outputs of |vertex|:
+            1. Let |maxFragmentInputLocation| be the greatest [=location=] value of all of the
+                user-defined inputs of |fragment|.
+            1. For each |location| from 0 to |maxFragmentInputLocation|:
+                1. If |vertex| does not define an output at |location|,
+                    |fragment| must not define an input at |location|.
+                1. If |vertex| does define an output at |location|,
+                    |fragment| must define an input at |location| with the same type and [=interpolation=].
```

Comment:
IIRC this matches what will be happening with D3D. However `@builtin(position)` is also something that must match between the vertex and fragment stage and it needs to be "somewhere". The group decided that it would be at the end of the inter-stage variables, such that if it is present in the fragment shader, all the inter stage variables MUST match.

Reply:
Aha, that was the main thing I didn't understand about the discussion on the issue. Thanks for the explanation!